code/game/objects/items/binoculars.dm

```diff
@@ -161,25 +172,64 @@
 	to_chat(user, "<span class='notice'>INITIATING LASER TARGETING. Stand still.</span>")
 	if(!do_after(user, max(1.5 SECONDS, target_acquisition_delay - (2.5 SECONDS * user.skills.getRating("leadership"))), TRUE, TU, BUSY_ICON_GENERIC) || world.time < laser_cooldown || laser)
 		return
-	if(mode)
-		var/obj/effect/overlay/temp/laser_coordinate/LT = new (TU, laz_name, S)
-		coord = LT
-		to_chat(user, "<span class='notice'>SIMPLIFIED COORDINATES OF TARGET. LONGITUDE [coord.x]. LATITUDE [coord.y].</span>")
-		playsound(src, 'sound/effects/binoctarget.ogg', 35)
-		while(coord)
-			if(!do_after(user, 50, TRUE, coord, BUSY_ICON_GENERIC))
-				QDEL_NULL(coord)
-				break
-	else
-		to_chat(user, "<span class='notice'>TARGET ACQUIRED. LASER TARGETING IS ONLINE. DON'T MOVE.</span>")
-		var/obj/effect/overlay/temp/laser_target/LT = new (TU, laz_name, S)
-		laser = LT
-		playsound(src, 'sound/effects/binoctarget.ogg', 35)
-		while(laser)
-			if(!do_after(user, 50, TRUE, laser, BUSY_ICON_GENERIC))
-				QDEL_NULL(laser)
-				break
-
+	switch(mode)
+		if(0)
```

Comment:
it might be work creating defines for the modes
```
#define MODE_CAS 0
#define MODE_RANGE_FINDER 1
#define MODE_RAILGUN 2
#define MODE_ORBITAL 3
```

Then you can undefine them at the bottom of the file.

Reply:
Since this has gotten quite large, it would be worth breaking out each mode into their own proc to make things a little easier to read & understand